### PR TITLE
Add DWARFv5 data representation

### PR DESCRIPTION
**Pull request by Luhrel**
Required for future DWARFv5 PRs

### CODE REVIEW

File: src/dmd/backend/dwarf2.d [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-/* Reflects declarations from the Dwarf 3 spec, not the The D Language Foundation
+/* Reflects declarations from the DWARF specification, not the The D Language Foundation
```
Comment (RazvanN7) [inline src/dmd/backend/dwarf2.d:2]: I think it would be useful to specify which Dwarf versions it supports (e.g Dwarf 3 to 5).

File: src/dmd/backend/dwarf2.d [hunk 2]
```diff
@@ -194,142 +258,198 @@ enum
...
-        DW_OP_breg1     = 0x71,
-        DW_OP_breg31    = 0x8f,
-        DW_OP_regx      = 0x90,
-        DW_OP_fbreg     = 0x91,
-        DW_OP_bregx     = 0x92,
-        DW_OP_piece     = 0x93,
-        DW_OP_deref_size        = 0x94,
-        DW_OP_xderef_size       = 0x95,
-        DW_OP_nop               = 0x96,
+        DW_OP_addr                      = 0x03,
```
Comment (RazvanN7) [inline src/dmd/backend/dwarf2.d:316]: Please do not mix refactorings with additions. It makes the code harder to review. This refactoring can be the subject of a different PR.

**Review by RazvanN7 — CHANGES_REQUESTED** [review]
Please do the refactorings in a different PR.

**Comment by Luhrel** [comment]
ping @RazvanN7. It should be easier to review now.

**Review by RazvanN7 — APPROVED** [review]

**Comment by Geod24** [comment]
Maybe this file shouldn't be named "dwarf2" anymore :D
